### PR TITLE
Integrated generated "ingest" client APIs into the existing module

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
Integrated generated ingest client APIs into the existing module, ensuring alignment with the server and maintaining backward compatibility

### Issues Resolved
Related to #477 

### Testing the "summary" parameter in the "get_pipeline" API resulted in an error indicating that the parameter is unrecognized (Implies generated "get_pipeline" is correct)
```
put https://localhost:9200/_ingest/pipeline/pipelineid123

{
  "description": "This pipeline processes student data",
  "processors": [
    {
      "set": {
        "description": "Sets the graduation year to 2023",
        "field": "grad_year",
        "value": 2023
      }
    },
    {
      "set": {
        "description": "Sets graduated to true",
        "field": "graduated",
        "value": true
      }
    },
    {
      "uppercase": {
        "field": "name"
      }
    }
  ]
}
```
{
"acknowledged": true
}

`get https://localhost:9200/_ingest/pipeline/pipelineid123?summary=true`
{
"error": {
"root_cause": [
{
"type": "illegal_argument_exception",
"reason": "request [/_ingest/pipeline/pipelineid123] contains unrecognized parameter: [summary]"
}
],
"type": "illegal_argument_exception",
"reason": "request [/_ingest/pipeline/pipelineid123] contains unrecognized parameter: [summary]"
},
"status": 400
}

### Ingest.geo_ip_stats should be removed from client(Generator is correct)

`client.ingest.geo_ip_stats() `
RequestError: RequestError(400, 'no handler found for uri [/_ingest/geoip/stats] and method [GET]', 'no handler found for uri [/_ingest/geoip/stats] and method [GET]')

### Before giving your approval, kindly double-check.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearchpy/_async/client/ingest.py [hunk 1]
```diff
@@ -25,42 +25,50 @@
...
+# https://github.com/opensearch-project/opensearch-api-specification/blob/main/OpenSearch.openapi.json
+# -----------------------------------------------------
+
+
 from .utils import SKIP_IN_PATH, NamespacedClient, _make_path, query_params
 
 
 class IngestClient(NamespacedClient):
-    @query_params("master_timeout", "cluster_manager_timeout", "summary")
+    @query_params("cluster_manager_timeout", "master_timeout")
```
Comment (dblock) [inline opensearchpy/_async/client/ingest.py:42]: I imagine this is correct?

**Review by dblock — APPROVED** [review]

File: opensearchpy/_async/client/ingest.py (hunk 1 above)
Comment (saimedhi) [reply opensearchpy/_async/client/ingest.py:42]: https://github.com/opensearch-project/opensearch-py/pull/513#issue-1916016963

File: opensearchpy/_async/client/ingest.py [hunk 2]
```diff
@@ -25,42 +25,50 @@
...
+        :arg id: Pipeline ID.
         :arg body: The ingest definition
-        :arg master_timeout (Deprecated: use cluster_manager_timeout): Explicit operation timeout for connection
-            to master node
-        :arg cluster_manager_timeout: Explicit operation timeout for connection
-            to cluster_manager node
-        :arg timeout: Explicit operation timeout
+        :arg cluster_manager_timeout: Operation timeout for connection
+            to cluster-manager node.
+        :arg master_timeout: Operation timeout for connection to master
```
Comment (VachaShah) [inline opensearchpy/_async/client/ingest.py:69]: May be we could keep the comments for master_timeout that it is deprecated?

Comment (saimedhi) [reply opensearchpy/_async/client/ingest.py:69]: @VachaShah, Do you mean include deprecation msg in the param description using generator.
something like 




:arg master_timeout (Deprecated: deprecation message from [here](https://github.com/opensearch-project/opensearch-api-specification/blob/main/OpenSearch.openapi.json#L8986) ): Explicit operation timeout for connection
            to master node

Comment (VachaShah) [reply opensearchpy/_async/client/ingest.py:69]: Yeah! It would be nice to have it in the description of the param.

**Review by VachaShah — APPROVED** [review]

**Pull request merged**